src/audio/fft/fft.c

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Amery Song <chao.song@intel.com>
+//	   Keyon Jie <yang.jie@linux.intel.com>
+
+#include <sof/common.h>
+#include "fft.h"
+#include "twiddle.h"
+
+static void icomplex32_add(struct icomplex32 *in1, struct icomplex32 *in2, struct icomplex32 *out)
+{
+	out->real = in1->real + in2->real;
+	out->image = in1->image + in2->image;
+}
+
+static void icomplex32_sub(struct icomplex32 *in1, struct icomplex32 *in2, struct icomplex32 *out)
+{
+	out->real = in1->real - in2->real;
+	out->image = in1->image - in2->image;
+}
+
+static void icomplex32_mul(struct icomplex32 *in1, struct icomplex32 *in2, struct icomplex32 *out)
+{
+	int64_t tmp;
+
+	tmp = (int64_t)in1->real * (int64_t)in2->real - (int64_t)in1->image * (int64_t)in2->image;
+	out->real = (int)(tmp >> 31);
+	tmp = (int64_t)in1->real * (int64_t)in2->image + (int64_t)in1->image * (int64_t)in2->real;
+	out->image = (int)(tmp >> 31);
+}
+
+void fft(struct icomplex32 *inb, struct icomplex32 *outb, unsigned int size)
+{
+	struct icomplex32 tmp1, tmp2;
+	int bitReverseIdx[4096] = {0};
+	int lim = 1, len = 0;
+	int i, j, k, m;
+	int depth;
+
+	while (lim < size) {
+		lim <<= 1;
+		len++;
+	}
+
+	for (i = 1; i < lim; ++i) {
+		bitReverseIdx[i] = (bitReverseIdx[i >> 1] >> 1) | ((i & 1) << (len - 1));
```

Comment:
Camelcase, should be bit_reverse_idx[]

Reply:
@singalsu Compared to implement a new FFT algorithm, I prefer to reuse FFT for IFFT, what do you think?

Reply:
Yes, that should make sense. I guess there are multiple ways, I saw e.g. conj(1/N * fft(conj(X)) might work. Based on quick try seemed to work. Also since it makes sense to optimize the inner loop of FFT with intrinsics the IFFT would share the same optimization work.